src/Compilers/Core/Portable/Syntax/SyntaxTree.cs

```diff
@@ -134,7 +134,9 @@ public Task<SyntaxNode> GetRootAsync(CancellationToken cancellationToken = defau
         /// <summary>
         /// Gets the root node of the syntax tree asynchronously.
         /// </summary>
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
         protected abstract Task<SyntaxNode> GetRootAsyncCore(CancellationToken cancellationToken);
```

Comment:
The name actually seems wrong to me.  Can it not be `GetRootCoreAsync`? Note: SyntaxTree has an internal constructor, so it's not a breaking change for us to change the names of protected members in it since they're effectively `internal and protected`.

Reply:
This is the internal method for public API `GetRootAsync`. It is typical to add a `Core` suffix for those. 
But that clashes with requiring an `Async` suffix ;-)

Reply:
> This is the internal method for public API `GetRootAsync`. It is typical to add a `Core` suffix for those.

Yes, we could modify our understanding of this situation such that between the two suffixes (`Async` and `Core`), `Core` gets placed before `Async`.